server/accounts.go

```diff
@@ -4230,15 +4295,23 @@ func (tr *transform) transform(tokens []string) (string, error) {
 	li := len(tr.dtpi) - 1
 	for i, index := range tr.dtpi {
 		// <0 means use destination token.
-		if index < 0 {
+		if index[0] < 0 {
 			token = tr.dtoks[i]
 			// Break if fwc
 			if len(token) == 1 && token[0] == fwc {
 				break
 			}
 		} else {
 			// >= 0 means use source map index to figure out which source token to pull.
-			token = tokens[index]
+			if tr.dtpinp[i] > 0 { // there is a valid (i.e. not -1) value for number of partitions, this is a partition transform token
+				var keyForHashing string
+				for _, sourceToken := range tr.dtpi[i] {
+					keyForHashing = keyForHashing + tokens[sourceToken]
```

Comment:
I'd assume most ppl only ever have one token to hash. 

However, because strings are immutable many objects are created and immediately discarded.
This is a nifty trick from our code base:
basically create a slice with an array as backing buffer.
append to it. (if the backing buffer is exceeded a bigger one will be allocated by append)

```
var (
  _ims           [64]byte
  ims            = _ims[:0]
)
for _, sourceToken := range tr.dtpi[i] {
    ims = append(ims, []byte(tokens[sourceToken]))
}
token = tr.getHashPartition(ims, int(tr.dtpinp[i]))
```

I assume getHashPartition signature changed to use []byte (which it internally casts to anyway)

Reply:
Thanks for the example code, will do that - some pretty advanced optimizing here 😮 but since in the function that gets call for every subject being transformed even small improvements do add up!